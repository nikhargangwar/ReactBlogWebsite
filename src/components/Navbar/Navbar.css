#innerheader {
    width: 70%;
    margin: 0 auto;
    height: 100%;

}

#innerheader h2 {
    margin-top: 20px;
    margin-bottom: 2px;
}

#innerheader h4 {
    margin-top: 5px;
}

#innerHeadContainer1 {
    display: flex;
    height: 70%;
    width: 100%;
    justify-content: center;
    flex-direction: column;
    align-items: flex-start;


}

#innerHeadContainer2 {
    height: 30%;
    width: 100%;
    display: flex;
    justify-content: flex-end;
}

.options {
    width: 40%;
    height: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;

}