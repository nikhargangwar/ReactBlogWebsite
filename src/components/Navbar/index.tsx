// import { useState } from 'react';
// import * as React from "react";

import './Navbar.css';

function Navbar() {
  return (
    <div id="innerheader">
      <div id="innerHeadContainer1">
        <h2>The Artifacts</h2>
        <h4>
          <i>Culture & Art blog</i>
        </h4>
      </div>
      <div color="black" id="innerHeadContainer2">
        <div className="options">
          <h5>Blog</h5>
          <h5>About</h5>
          <h5>Contact</h5>
        </div>
      </div>
    </div>
  );
}

export default Navbar;
