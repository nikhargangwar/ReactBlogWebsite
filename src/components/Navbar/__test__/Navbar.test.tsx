import React from 'react';
import { getByTestId, render } from '@testing-library/react';
import Navbar from '../../Navbar';

describe('Header', () => {
  it('should render when component is mounted', () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
  it('should have its child components rendered perfectly',()=>{
    const {getByText} = render(<Navbar />);
    const childComponent = getByText('About')
    expect(childComponent).toBeTruthy();
  });

  it('should have option tag rendered',()=>{
    const {getByTestId} = render(<Navbar />);
    const childComponent = getByTestId('custom-element');
    expect(childComponent).toBeTruthy();
  });

});
