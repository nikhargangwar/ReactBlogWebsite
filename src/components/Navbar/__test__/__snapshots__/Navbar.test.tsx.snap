// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render when component is mounted 1`] = `
<div>
  <div
    class="header"
  >
    <div
      id="innerheader"
    >
      <div
        id="innerHeadContainer1"
      >
        <h2>
          The Artifacts
        </h2>
        <h4>
          <i>
            Culture & Art blog
          </i>
        </h4>
      </div>
      <div
        color="black"
        id="innerHeadContainer2"
      >
        <div
          class="options"
          data-testid="custom-element"
        >
          <h5>
            Blog
          </h5>
          <h5>
            About
          </h5>
          <h5>
            Contact
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
